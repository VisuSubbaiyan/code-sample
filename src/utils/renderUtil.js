import React from 'react';

import {Airport, BoardingPass, Ground} from '../components';

export const renderItems = items => items.map(item => {
  const { type, segment_id } = item;

  switch (type) {
    case 'GROUND': {
      return <Ground {...item} key={segment_id} className="segment-active no-top-padding" />;
    }
    
    case 'AIRPORT': {
      return <Airport {...item} key={segment_id} />;
    }

    default:
      return <BoardingPass {...item} key={segment_id} />
  }
});
