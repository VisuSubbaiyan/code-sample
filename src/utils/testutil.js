import { mount, shallow } from 'enzyme';
import { prettyPrint } from 'html';

export const mockFunction = jest.fn(value => value);

export const sampleFriend = {
  name: 'Sample friend',
  gender: 'male'
};

export const sampleId = 1;

export const createMountedComponent = ({ children, isShallow }) => {
  const mountedComponent = isShallow ? shallow(children) : mount(children);
  mountedComponent.htmlRender = () => prettyPrint(mountedComponent.html(), { indent_size: 2 });

  return mountedComponent;
};

