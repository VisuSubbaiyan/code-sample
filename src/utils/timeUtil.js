import moment from 'moment';

export const formatTime = time => {
  const tempTime = moment.duration(time);
  
  return `${tempTime.hours()}: ${tempTime.minutes()}`;
}