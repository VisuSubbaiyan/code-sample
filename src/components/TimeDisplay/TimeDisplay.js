import React from 'react';

import { formatTime } from '../../utils/timeUtil';

const TimeDisplay = props => <h3 className="time">{formatTime(props.time)}</h3>;

export default TimeDisplay;
