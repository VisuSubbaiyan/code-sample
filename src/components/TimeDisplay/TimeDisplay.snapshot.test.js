import React from 'react';

import { createMountedComponent } from '../../utils/testutil';

import TimeDisplay from './TimeDisplay';

describe('<TimeDisplay /> should', () => {
  const mountedComponent = createMountedComponent({
    children: <TimeDisplay time={1523596800} />
  });

  it('render correctly according to props', () => {
    expect(mountedComponent.htmlRender()).toMatchSnapshot();
  })
});