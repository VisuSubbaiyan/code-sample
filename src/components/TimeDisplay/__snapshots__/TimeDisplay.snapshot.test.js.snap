// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeDisplay /> should render correctly according to props 1`] = `"<h3 class=\\"time\\">15: 13</h3>"`;
